src/button-dropdown/internal.tsx

```diff
@@ -149,11 +149,7 @@ const InternalButtonDropdown = React.forwardRef(
         >
           {hasHeader && (
             <div className={styles.header} id={headerId}>
-              {title && (
-                <InternalBox fontSize="heading-s" fontWeight="bold">
-                  <span className={styles.title}>{title}</span>
-                </InternalBox>
-              )}
+              {title && <h3 className={styles.title}>{title}</h3>}
```

Comment:
To avoid being overridden by any global styles: 

```suggestion
                <div className={styles.title}>
                  <InternalBox
                    fontSize="heading-s"
                    fontWeight="bold"
                    color="inherit"
                    tagOverride="h3"
                    margin={{ vertical: 'n' }}
                  >
                    {title}
                  </InternalBox>
                </div>
```